### PR TITLE
Providers Tutorial

### LINKED ISSUE #6076: Providers Tutorial

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Closes #6076

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by nikoulai — APPROVED** [review]

File: docs/docs/tutorials/providers.md [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+
+5. Create a new JavaScript file called `web3-ipc-provider.js` in your code editor.
+6. Copy and paste the following code into your `web3-ipc-provider.js` file and save it:
+
+    ```js
+    const { Web3 } = require('web3');
+
+    // Connect to the Ethereum network using IPC provider
+    const ipcPath = '<path>';
+    const ipcProvider = new Web3.providers.IpcProvider(ipcPath);
```
Comment (jdevcs) [inline docs/docs/tutorials/providers.md]: this code will need update as IPC is not exported through main web3 package now.

**Comment by jdevcs** [comment]
Thanks @Muhammad-Altabba for your contribution.

**Review by avkos — APPROVED** [review]

File: docs/docs/guides/web3_migration_guide/providers_migration_guide.md [hunk 2]
```diff
@@ -245,7 +245,7 @@ You can use it by installing `web3-providers-ipc` and creating a new instance. S
 you can pass it on to the Web3 instance.
 
 ```ts
-import IpcProvider from 'web3-providers-ipc';
+import { IpcProvider } from 'web3-providers-ipc';
```
Comment (spacesailor24) [inline docs/docs/guides/web3_migration_guide/providers_migration_guide.md:248]: What's the reason for this change? Our other providers (http and ws) import via the default, and [IpcProvider](https://github.com/web3/web3.js/blob/3b95b5ef52fcda4c3e40066a2cf54f9264389fab/packages/web3-providers-ipc/src/index.ts#L62) is still the default export

Comment (Spacesai1or) [inline docs/docs/guides/web3_migration_guide/providers_migration_guide.md:248]: What's the reason for this change? Our other providers (http and ws) import via the default, and [IpcProvider](https://github.com/web3/web3.js/blob/3b95b5ef52fcda4c3e40066a2cf54f9264389fab/packages/web3-providers-ipc/src/index.ts#L62) is still the default export

File: docs/docs/tutorials/providers.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+
+4. Create a new JavaScript file called `web3-http-provider.js` in your code editor.
+
+5. Copy and paste the following code into your `web3-http-provider.js` file and save it:
+
+    ```js
+    const { Web3 } = require('web3');
+
+    // Connect to the Ethereum network using the HTTP provider
+    const ganacheUrl = 'http://localhost:7545';
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md:71]:
```suggestion
    const ganacheUrl = 'http://localhost:8545';
```

File: docs/docs/tutorials/providers.md [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+    main();
+    ```
+
+6. Ensure that Ganache is running as mentioned in the [Prerequisites](#prerequisites) section.
+
+7. In the command prompt or terminal window, type `node web3-http-provider.js` and press Enter. This will run your JavaScript file and connect to the Ethereum network using the HTTP provider and Ganache.
+
+If everything is set up properly, you should see the current block number, a transaction hash, transaction receipt, and the updated block number printed in the console.
+
+This example showed how to connect to the Ethereum network using HTTP provider, get the current block number, send a transaction, and get the updated block number.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
If everything is set up properly, you should see the current block number, a transaction hash, transaction receipt, and the updated block number printed in the console:

```bash
Current block number: 0n
Transaction Receipt: {
  transactionHash: '0x0578672e97d072b4b91773c8bfc710e4f777616398b82b276323408e59d11362',
  transactionIndex: 0n,
  blockNumber: 1n,
  blockHash: '0x348a6706e7cce6547fae2c06b3e8eff1f58e4669aff88f0af7ca250ffdcdeef5',
  from: '0x6e599da0bff7a6598ac1224e4985430bf16458a4',
  to: '0x6f1df96865d09d21e8f3f9a7fba3b17a11c7c53c',
  cumulativeGasUsed: 21000n,
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  effectiveGasPrice: 2000000000n,
  type: 0n
}
Updated block number: 1n
```

This example showed how to connect to the Ethereum network using HTTP provider, get the current block number, send a transaction, and get the updated block number.
```

Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: Formatting looks weird because Github's suggestions uses ` ```suggestion `, but I added the expected output:

```bash
Current block number: 0n
Transaction Receipt: {
  transactionHash: '0x0578672e97d072b4b91773c8bfc710e4f777616398b82b276323408e59d11362',
  transactionIndex: 0n,
  blockNumber: 1n,
  blockHash: '0x348a6706e7cce6547fae2c06b3e8eff1f58e4669aff88f0af7ca250ffdcdeef5',
  from: '0x6e599da0bff7a6598ac1224e4985430bf16458a4',
  to: '0x6f1df96865d09d21e8f3f9a7fba3b17a11c7c53c',
  cumulativeGasUsed: 21000n,
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  effectiveGasPrice: 2000000000n,
  type: 0n
}
Updated block number: 1n
```

File: docs/docs/tutorials/providers.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+
+4. Create a new JavaScript file called `web3-websocket-provider.js` in your code editor.
+
+5. Copy and paste the following code into your `web3-websocket-provider.js` file and save it:
+
+    ```js
+    const { Web3 } = require('web3');
+
+    // Connect to the Ethereum network using the WebSocket provider
+    const ganacheUrl = 'ws://localhost:7545';
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
    const ganacheUrl = 'ws://localhost:8545';
```

File: docs/docs/tutorials/providers.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+    const wsProvider = new Web3.providers.WebsocketProvider(ganacheUrl);
+    const web3 = new Web3(wsProvider);
+
+    async function main() {
+    	try {
+    		console.log(
+    			'Do the provider supports subscription?:',
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
    			'Does the provider support subscriptions?:',
```

File: docs/docs/tutorials/providers.md [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+    main();
+    ```
+
+6. Ensure that Ganache is running as mentioned in the [Prerequisites](#prerequisites) section.
+
+7. Type `node web3-websocket-provider.js` in the command prompt or terminal window and press Enter. This will run your JavaScript file.
+
+If everything is set up properly, you should see the new block headers, transaction hash, and pending transaction printed in the console. The unique feature of WebSocket provider highlighted in this example is that it can subscribe to new block headers and pending transactions in real-time.
+
+### IPC Provider (for Node.js)
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
If everything is set up properly, you should see the new block headers, transaction hash, and pending transaction printed in the console. The unique feature of WebSocket provider highlighted in this example is that it can subscribe to new block headers and pending transactions in real-time:

```bash
Does the provider support subscriptions?: true
Transaction Receipt: {
  transactionHash: '0x0578672e97d072b4b91773c8bfc710e4f777616398b82b276323408e59d11362',
  transactionIndex: 0n,
  blockNumber: 1n,
  blockHash: '0x5c05248fe0fb8f45a8c9b9600904a36c0e5c74dce01495cfc72278c185fe7838',
  from: '0x6e599da0bff7a6598ac1224e4985430bf16458a4',
  to: '0x6f1df96865d09d21e8f3f9a7fba3b17a11c7c53c',
  cumulativeGasUsed: 21000n,
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  effectiveGasPrice: 2000000000n,
  type: 0n
}
```

### IPC Provider (for Node.js)
```

Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: Added:

```bash
Does the provider support subscriptions?: true
Transaction Receipt: {
  transactionHash: '0x0578672e97d072b4b91773c8bfc710e4f777616398b82b276323408e59d11362',
  transactionIndex: 0n,
  blockNumber: 1n,
  blockHash: '0x5c05248fe0fb8f45a8c9b9600904a36c0e5c74dce01495cfc72278c185fe7838',
  from: '0x6e599da0bff7a6598ac1224e4985430bf16458a4',
  to: '0x6f1df96865d09d21e8f3f9a7fba3b17a11c7c53c',
  cumulativeGasUsed: 21000n,
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  effectiveGasPrice: 2000000000n,
  type: 0n
}
```

File: docs/docs/tutorials/providers.md [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+
+    Node.js is a JavaScript runtime environment that enables you to run JavaScript on the server-side. You can download it from [https://nodejs.org/en/download/](https://nodejs.org/en/download/).
+
+3. **npm**
+
+    npm (Node Package Manager) is used to publish and install packages to and from the public npm registry or a private npm registry. You can install it by following the instructions here: [https://docs.npmjs.com/downloading-and-installing-node-js-and-npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+    Alternatively, you can use **yarn** instead of **npm** by following the instructions here: [https://classic.yarnpkg.com/lang/en/docs/getting-started/](https://classic.yarnpkg.com/lang/en/docs/getting-started/).
+
+## Types of Providers
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
3. **npm**

    npm (Node Package Manager) is used to publish and install packages to and from the public npm registry or a private npm registry. You can install it by following the instructions here: [https://docs.npmjs.com/downloading-and-installing-node-js-and-npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).

    Alternatively, you can use **yarn** instead of **npm** by following the instructions here: [https://classic.yarnpkg.com/lang/en/docs/getting-started/](https://classic.yarnpkg.com/lang/en/docs/getting-started/).
    
4. **Geth** (Optional, used for IPC provider example)

    Geth (go-ethereum) is an Ethereum execution client meaning it handles transactions, deployment and execution of smart contracts and contains an embedded computer known as the Ethereum Virtual Machine. You can install it by following the instructions here: [https://geth.ethereum.org/docs/getting-started/installing-geth](https://geth.ethereum.org/docs/getting-started/installing-geth)

## Types of Providers
```

File: docs/docs/tutorials/providers.md [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+
+7. Type `node web3-websocket-provider.js` in the command prompt or terminal window and press Enter. This will run your JavaScript file.
+
+If everything is set up properly, you should see the new block headers, transaction hash, and pending transaction printed in the console. The unique feature of WebSocket provider highlighted in this example is that it can subscribe to new block headers and pending transactions in real-time.
+
+### IPC Provider (for Node.js)
+
+The IPC Provider allows you to connect to an Ethereum node using Inter-Process Communication (IPC) in a Node.js environment. This provider is useful when you have a local Ethereum node running on your machine and want to interact with it using Node.js.
+
+In the following steps you will run `geth` in development mode and you will run a code that reads geth accounts and send a transaction:
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
In the following steps you will run `geth` in development mode and you will run code that reads Ethereum accounts and sends a transaction:
```

File: docs/docs/tutorials/providers.md [hunk 10]
```diff
@@ -0,0 +1,469 @@
...
+    }
+
+    main();
+    ```
+
+7. replace `<path>` with the `ipcPath` that you had specified, when starting the `geth` node, in the first step.
+
+    **Note**: replace `<path>` with the `ipcPath` specified when starting the `geth` node.
+
+8. Type `node web3-ipc-provider.js` in the command prompt or terminal window and press Enter. This will run your JavaScript file.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
7. replace `<path>` with the `ipcPath` that you had specified, when starting the `geth` node, in the first step.

8. Type `node web3-ipc-provider.js` in the command prompt or terminal window and press Enter. This will run your JavaScript file.
```

File: docs/docs/tutorials/providers.md [hunk 11]
```diff
@@ -0,0 +1,469 @@
...
+    		);
+
+    		// Get the list of accounts in the connected node which is in this case: geth in dev mode.
+    		const accounts = await web3.eth.getAccounts();
+    		console.log('Accounts:', accounts);
+
+    		// Send a transaction to the network
+    		const transactionReceipt = await web3.eth.sendTransaction({
+    			from: accounts[0],
+    			to: accounts[1],
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
    			to: accounts[0],
```

Geth creates a single address by default. using `accounts[1]` for `to` results in `undefined` causing an error: `An error occurred: InvalidResponseError: Returned error: contract creation without any data provided`

File: docs/docs/tutorials/providers.md [hunk 12]
```diff
@@ -0,0 +1,469 @@
...
+
+7. replace `<path>` with the `ipcPath` that you had specified, when starting the `geth` node, in the first step.
+
+    **Note**: replace `<path>` with the `ipcPath` specified when starting the `geth` node.
+
+8. Type `node web3-ipc-provider.js` in the command prompt or terminal window and press Enter. This will run your JavaScript file.
+
+If everything is set up properly, you should see the list of accounts and transaction receipt printed in the console.
+
+Keep in mind that using IPC Provider with `geth` in development mode in a production environment is not recommended as it can pose a security risk.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
If everything is set up properly, you should see the list of accounts and transaction receipt printed in the console:

```bash
Do the provider supports subscription?: true
Accounts: [ '0x82333ED0FAA7a883297C4d8e0FDE1E1CFABAeB7D' ]
Transaction Receipt: {
  blockHash: '0xd1220a9b6f86083e420da025179593f5aad3732165a687019a89528a4ab2bcd8',
  blockNumber: 1n,
  cumulativeGasUsed: 21000n,
  effectiveGasPrice: 1000000001n,
  from: '0x82333ed0faa7a883297c4d8e0fde1e1cfabaeb7d',
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  to: '0x82333ed0faa7a883297c4d8e0fde1e1cfabaeb7d',
  transactionHash: '0x76c05df78dc5dbfade0d11322b3cadc894c17efe36851856aca29488b47c3fbd',
  transactionIndex: 0n,
  type: 0n
}
```

Keep in mind that using IPC Provider with `geth` in development mode in a production environment is not recommended as it can pose a security risk.
```

Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: Added:

```bash
Do the provider supports subscription?: true
Accounts: [ '0x82333ED0FAA7a883297C4d8e0FDE1E1CFABAeB7D' ]
Transaction Receipt: {
  blockHash: '0xd1220a9b6f86083e420da025179593f5aad3732165a687019a89528a4ab2bcd8',
  blockNumber: 1n,
  cumulativeGasUsed: 21000n,
  effectiveGasPrice: 1000000001n,
  from: '0x82333ed0faa7a883297c4d8e0fde1e1cfabaeb7d',
  gasUsed: 21000n,
  logs: [],
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  status: 1n,
  to: '0x82333ed0faa7a883297c4d8e0fde1e1cfabaeb7d',
  transactionHash: '0x76c05df78dc5dbfade0d11322b3cadc894c17efe36851856aca29488b47c3fbd',
  transactionIndex: 0n,
  type: 0n
}
```

File: docs/docs/tutorials/providers.md [hunk 13]
```diff
@@ -0,0 +1,469 @@
...
+
+4. Create a new JavaScript file, called `index.js`, in your code editor.
+5. Copy and paste the following code into your JavaScript file, and then save the file:
+
+    ```js
+    const { Web3 } = require('web3');
+    const HDWalletProvider = require('@truffle/hdwallet-provider');
+    const bip39 = require('bip39');
+
+    const mnemonic = bip39.generateMnemonic(); //generates seed phrase
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
    const mnemonic = bip39.generateMnemonic(); // generates seed phrase
```

File: docs/docs/tutorials/providers.md [hunk 14]
```diff
@@ -0,0 +1,469 @@
...
+    	.getBlockNumber()
+    	.then(function (blockNumber) {
+    		console.log('Current block number:', blockNumber);
+    	})
+    	.catch(function (error) {
+    		console.log('Error:', error);
+    	});
+    ```
+
+6. Replace `'YOUR_INFURA_PROJECT_ID'` with your own Infura project ID. You can obtain an Infura project ID by signing up for a free account at https://infura.io/register. Alternatively, you can use any other URL that is compatible with HDWalletProvider, such as a local Ganache accessible at `'http://localhost:7545'`.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md:405]:
```suggestion
6. Replace `'YOUR_INFURA_PROJECT_ID'` with your own Infura project ID. You can obtain an Infura project ID by signing up for a free account at https://infura.io/register. Alternatively, you can use any other URL that is compatible with HDWalletProvider, such as a local Ganache accessible at `'http://localhost:8545'`.
```

File: docs/docs/tutorials/providers.md [hunk 15]
```diff
@@ -0,0 +1,469 @@
...
+
+7. In the command prompt, run `node index.js` and press Enter. This will execute your JavaScript file and connect to the Ethereum network using HDWalletProvider with Infura.
+
+If everything is set up properly, you should see the current block number printed in the console.
+
+This code will connect your web application to the Ethereum network using truffle HD Wallet-enabled Web3 provider. You can modify it to interact with the network, perform transactions, and read/write data from the Ethereum network.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]:
```suggestion
7. In the command prompt, run `node index.js` and press Enter. This will execute your JavaScript file and connect to the Ethereum network using HDWalletProvider with Infura.

If everything is set up properly, you should see the current block number printed in the console:

:::danger
Your seed phrase gives complete access to your Ethereum account and it should **never** be shared with anyone you don't want to give full access to your account. The seed phrase is `console.log`ed in the code example to show you what it looks like, but you should **never** do this with a seed phrase to an account you plan on using to send real money
:::

```bash
seed phrase: remain climb clock valid budget cable tunnel force split level measure repair
Current block number: 17317844n
```

This code will connect your web application to the Ethereum network using truffle HD Wallet-enabled Web3 provider. You can modify it to interact with the network, perform transactions, and read/write data from the Ethereum network.
```

Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: Added:

```bash
:::danger
Your seed phrase gives complete access to your Ethereum account and it should **never** be shared with anyone you don't want to give full access to your account. The seed phrase is `console.log`ed in the code example to show you what it looks like, but you should **never** do this with a seed phrase to an account you plan on using to send real money
:::

seed phrase: remain climb clock valid budget cable tunnel force split level measure repair
Current block number: 17317844n
```

File: docs/docs/tutorials/providers.md [hunk 16]
```diff
@@ -0,0 +1,469 @@
...
+    		<title>Connecting to the Ethereum network with Web3.js and MetaMask</title>
+    	</head>
+    	<body>
+    		<h1>Connecting to the Ethereum network with Web3.js and MetaMask</h1>
+    		<div id="log">
+    			You need to approve connecting this website to MetaMask. Click on the MetaMask icon
+    			in the browser extension, if it did not show a popup already.
+    		</div>
+
+    		<script src="https://cdn.jsdelivr.net/npm/web3@4.0.1-rc.1/dist/web3.min.js"></script>
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]: This doesn't currently exist

File: docs/docs/tutorials/providers.md [hunk 17]
```diff
@@ -0,0 +1,469 @@
...
+
+    app.listen(8000, () => {
+    	console.log('Server started on port 8000');
+    });
+    ```
+
+5. Save the file.
+6. Open your command prompt window and navigate to the directory where `server.js` is located.
+7. Type `node server.js` and press Enter. This will start the server on port 8000.
+8. Open your web browser and navigate to `http://localhost:8000/`. MetaMask should ask for your approval to connect to your website. Follow the steps and give your consent.
```
Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md]: We could also just have the user open the HTML file in their browser and forego having to install and setup express

**Review by Spacesai1or — CHANGES_REQUESTED** [review]

File: docs/docs/guides/web3_migration_guide/providers_migration_guide.md (hunk 2 above)
Comment (Muhammad-Altabba) [reply docs/docs/guides/web3_migration_guide/providers_migration_guide.md:248]: Actually, there was a comment in another MR from Marin that default exports are not good. There are many articles about why default export is bad or is not preferred. You may check for example: https://basarat.gitbook.io/typescript/main-1/defaultisbad.

File: docs/docs/tutorials/providers.md (hunk 3 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md:71]: Actually, the example uses Ganache application which runs on `7545` by default.

File: docs/docs/tutorials/providers.md (hunk 5 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md]: Actually, the example uses Ganache application which runs on `7545` by default.

File: docs/docs/tutorials/providers.md (hunk 17 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md]: Unfortunately, this would not work because of browser security restrictions. 
(I discovered that I had a very old question about this https://github.com/MetaMask/metamask-extension/issues/6162 :smile:)

File: docs/docs/tutorials/providers.md (hunk 14 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md:405]: Actually, the example uses Ganache application which runs on `7545` by default.

File: docs/docs/guides/web3_migration_guide/providers_migration_guide.md (hunk 2 above)
Comment (Spacesai1or) [reply docs/docs/guides/web3_migration_guide/providers_migration_guide.md:248]: @Muhammad-Altabba Then we should update our use of default exports for everything all at once, and not just one at a time - this could confuse users

File: docs/docs/tutorials/providers.md [hunk 18]
```diff
@@ -0,0 +1,613 @@
...
+  logs: [],
+  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
+  status: 1n,
+  effectiveGasPrice: 2000000000n,
+  type: 0n
+}
+Updated block number: 1n
+```
+
+### WebSocket Provider
```
Comment (spacesailor24) [inline docs/docs/tutorials/providers.md:129]:
```suggestion
```

This example showed how to connect to the Ethereum network using HTTP provider, get the current block number, send a transaction, and get the updated block number.

### WebSocket Provider
```

Comment (Spacesai1or) [inline docs/docs/tutorials/providers.md:130]:
```suggestion

This example showed how to connect to the Ethereum network using HTTP provider, get the current block number, send a transaction, and get the updated block number.

### WebSocket Provider
```

File: docs/docs/guides/web3_migration_guide/providers_migration_guide.md (hunk 2 above)
Comment (Spacesai1or) [reply docs/docs/guides/web3_migration_guide/providers_migration_guide.md:248]: https://github.com/web3/web3.js/issues/6116

File: docs/docs/tutorials/providers.md (hunk 16 above)
Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: https://github.com/web3/web3.js/issues/6117

File: docs/docs/tutorials/providers.md (hunk 10 above)
Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md]: I removed

```
**Note**: replace `<path>` with the `ipcPath` specified when starting the `geth` node.
```

because it's repeating the information given right above it

File: docs/docs/tutorials/providers.md (hunk 18 above)
Comment (Spacesai1or) [reply docs/docs/tutorials/providers.md:130]: You had this block of text before, but I think it got mistakenly removed with the edits

File: docs/docs/tutorials/providers.md (hunk 10 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md]: Thanks, I forgot to remove it.

File: docs/docs/tutorials/providers.md (hunk 18 above)
Comment (Muhammad-Altabba) [reply docs/docs/tutorials/providers.md:130]: Actually, I removed it as it is a repetition of what is mentioned here: 
> To connect to the Ethereum network using the HTTP provider, follow these steps:

And here:
> If everything is set up properly, you should see the current block number, the transaction receipt, and the updated block number printed in the console:

Thanks,

**Review by Spacesai1or — APPROVED** [review]
